tests/integration/test_charm_runner.py

```diff
@@ -155,13 +161,6 @@ async def test_flush_runner_and_resource_config(
         assert action.status == "completed"
         assert action.results["delta"]["virtual-machines"] == "0"
 
-        await wait_for(lambda: workflow.update() or workflow.status == "completed")
-        action = await app.units[0].run_action("flush-runners")
-        await action.wait()
-
-        assert action.status == "completed"
-        assert action.results["delta"]["virtual-machines"] == "1"
```

Comment:
This is not necessary to test (and its brittle, because flush runners could still be run on an "idle" runner, we would need to wait_for the runner to be considered offline/idle). I have added tear down code to the fixture above to ensure that a subsequent test is not affected by this test.